### PR TITLE
G2 2023 v3 issue#13401

### PR DESCRIPTION
**Pull request by c21axepe**
Created a txt file with tests for the services in showDuggaservice.php
I might have missed something so feedback is always welcomed!

### CODE REVIEW

**Comment by c21axepe** [comment]
Created a txt file with tests for the services in showDuggaservice.php
I might have missed somthing so feedback is always welcomed!

**Review by a21oscgu — CHANGES_REQUESTED** [review]
Looks good, but:
Row 77 - did you mean "test" instead of "text"?
Row 83 - "this part", what are you referring to?
Row 88 - did you mean "successful" instead of "succeful"?

**Comment by c21axepe** [comment]
Made changes as requested!

File: backend-models/service-tests/showDuggaservice-tests.txt [hunk 1]
```diff
@@ -0,0 +1,94 @@
+showDuggaservice.php
+
+-------------------------------------------------------------------------------------------------------------
+TEST#1
+updateActiveUsersTest(100-118)
+Test to ensure active users can be updated/created.
+First make a sql query that creates an active_users in groupdugga. Then try updating the active_users with the newly created values.
+Or try updating an active users without a active_users token. Then it will create a AUtoken for that active users hash
+
+Pre-req: {($opt, "UPDATEAU")==0}
```
[the remark this answers is not in the document]
Comment (c21axepe) [reply backend-models/service-tests/showDuggaservice-tests.txt:10]: Aaaah ok i understand, will look in to this right away!

File: backend-models/service-tests/showDuggaservice-tests.txt [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+SQL query to get active_users: SELECT active_users FROM groupdugga WHERE hash=hash;
+hash gets binded to $hash that is then executed
+active_users gets fetched and $active gets the value of active_users
+
+SQL query: If no active_users is found: INSERT INTO groupdugga(hash,active_users) VALUES(hash,AUtoken);
+hash gets bind to $hash
+AUtoken gets bind to $AUtoken
+Its the executed
+
```
[the remark this answers is not in the document]
Comment (a19hakja) [reply backend-models/service-tests/showDuggaservice-tests.txt:20]: This includes researching what values are valid, what values already are in the database (with for example, mySQL workbench).

Make sure to follow the README for this issue. @a21jeaha made the first template and can probably provide you with more details of his research about what's needed here.

**Review by a21oscgu — APPROVED** [review]
I see no errors

**Comment by b21kurar** [comment]
The purpose of this issue is to ensure that all values have been decided upon **before** the test is written, therefore I would like to ask two things:

1.  Why is there no values for expected output? This should preferably, be an actual value to be able to compare. (Maybe it isn't possible in your case?)
2. Are the SQL querys still relevant? The `send` values should let the service handle all database queries. Are there any reasons the SQL queries are included in the test cases?

**Comment by d21erisu** [comment]
> The purpose of this issue is to ensure that all values have been decided upon **before** the test is written, therefore I would like to ask two things:
> 
> 1. Why is there no values for expected output? This should preferably, be an actual value to be able to compare. (Maybe it isn't possible in your case?)
> 2. Are the SQL querys still relevant? The `send` values should let the service handle all database queries. Are there any reasons the SQL queries are included in the test cases?

@c21axepe

**Pull request closed without merging**